mitodl/ocw-studio
# feat: unpublish sites

#### Pre-Flight checklist


#### What are the relevant tickets?
closes https://github.com/mitodl/ocw-studio/issues/1642

#### What's this PR do?
- Allow course authors to unpublish websites
- Introduces a dropdown to the site listing view
    - Since this dropdown has only one `Unpublish` option so the dropdown is not shown is the website is `draft`  
- Upon clicking `Unpublish` button, a modal is opened with confirmation.
- Upon confirming, the site is unpublished (unpublish pipeline is triggered)

#### How should this be manually tested?
- Checkout this branch
- Go to the site listing view.
- Verify that the dropdown with a `Unpublish` option is being shown only with sites that are published, and dropdown is not being shown for draft websites. 
- Verify that the dropdown works fine.
- Verify that upon clicking `unpublish`, a modal is opened asking the user for a confirmation.
- Verify that the modal works fine. Closes upon clicking close/x button
- Verify that, upon clicking `Confirm` button, the site is successfully unpublished (unpublish pipeline is triggered)

#### Screenshots (if appropriate)

<img width="1395" alt="image" src="https://user-images.githubusercontent.com/93309234/219381297-d207591c-ec89-4ddb-b760-7bbaf1a6bb08.png">

<img width="247" alt="image" src="https://user-images.githubusercontent.com/93309234/219381348-de998a1b-4316-4315-a899-89e66fa20667.png">

<img width="445" alt="image" src="https://user-images.githubusercontent.com/93309234/220342233-3169d606-49a3-425b-8bb7-27dd6c78b366.png">

<img width="551" alt="image" src="https://user-images.githubusercontent.com/93309234/219698947-c155aae0-bebc-4a42-871f-ec8de6942b89.png">
